pkg/k8s/apis/cilium.io/v2alpha1/bgpp_types.go

```diff
@@ -9,6 +9,12 @@ import (
 	slimv1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/apis/meta/v1"
 )
 
+const (
+	// DefaultPeerPort is the TCP port number of a CiliumBGPNeighbor when
+	// PeerPort is unspecified.
+	DefaultPeerPort uint16 = 179
```

Comment:
I think we are defining defaults here as well : 
https://github.com/cilium/cilium/blob/main/pkg/bgpv1/agent/controller.go#L430 

I don't have any objection on defining defaults in CRD definitions here, as long as we are consistent how we do it. 
cc @rastislavs

Reply:
In general I think defaulting in CRD should be preferred (including below `+kubebuilder:default=179`), but in some cases it is not possible as some default values are not constants - e.g. default `KeepAliveTime = HoldTime / 3`. So not exactly sure how we should do it to be consistent.

Reply:
From a code visibility point of view, it's better to set them at once in a single place. However, using kube-apiserver for setting the default sounds like a good idea. Maybe we can set the default at runtime for now and refactor them to use apiserver later on?

Reply:
If we agree with setting default at runtime, we can remove all `SetDefaultPeerPort`.

Reply:
> Maybe we can set the default at runtime for now and refactor them to use apiserver later on?

+1 on refactoring defaulting to use apiserver later for all fields (where it is possible) at once, I volunteer to do that refactor ;)

Reply:
In other projects that I've worked on, specifying default field values is usually associated with the associated API type. This allows the default value to be versioned with the API.

Reply:
I've created https://github.com/cilium/cilium/issues/25871 to track the need to move default values to the API type definitions.

Reply:
> If we agree with setting default at runtime, we can remove all SetDefaultPeerPort.

`SetDefaultPeerPort()` still needs to be called from `applyPolicyDefaults()` to set the default peer port.